API/Controllers/SearchController.cs

```diff
@@ -0,0 +1,58 @@
+using API.DTOs;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Authorize(Roles = "system")] // Internal access only at this time
+[AllowAnonymous]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class SearchController(ITournamentsService tournamentsService, IMatchesService matchesService, IPlayerService playerService) : Controller
+{
+    [HttpPost]
```

Comment:
Make sure to document this with `EndpointSummaryAttribute` and `EndpointDescriptionAttribute` for clarity. I usually use EndpointSummary for a very brief 1 sentence description, and include an EndpointDescription for more in-depth info.

Reply:
Oops, I did this for other endpoints I made just completely forgot to add them here